src/__private_api.rs

```diff
@@ -1,57 +1,116 @@
 //! WARNING: this is not part of the crate's public API and is subject to change at any time
 
+use self::sealed::KVs;
 use crate::{Level, Metadata, Record};
 use std::fmt::Arguments;
-pub use std::option::Option;
 pub use std::{file, format_args, line, module_path, stringify};
 
+#[cfg(feature = "kv_unstable")]
+pub type Value<'a> = dyn crate::kv::value::ToValue + 'a;
+
 #[cfg(not(feature = "kv_unstable"))]
-pub fn log(
+pub type Value<'a> = str;
+
+mod sealed {
+    /// Argument that provides key-value pairs.
+    pub trait KVs<'a> {
+        fn into_kvs(self) -> Option<&'a [(&'a str, &'a super::Value<'a>)]>;

```

Comment:
Why not take "&self" so that it can be used as a trait object?

```suggestion
        fn into_kvs(&self) -> Option<&'a [(&'a str, &'a super::Value<'a>)]>;

```

Reply:
Normally, I prefer passing small `Copy` types by value instead of reference. Clippy has a check for this: <https://rust-lang.github.io/rust-clippy/master/index.html#/trivially_copy_pass_by_ref>. Also, this is a private trait, all usages are restricted to this crate, and it is not used as a trait object, so it is unnecessary to make it object safe.

Reply:
It does make sense to pass kvs in value instead of reference, resolving this.